src/NLog/Config/ServiceRepository.cs

```diff
@@ -53,6 +53,8 @@ public abstract class ServiceRepository : IServiceProvider
         /// <remarks>Avoid calling this while handling a LogEvent, since random deadlocks can occur.</remarks>
         public abstract object GetService(Type serviceType);
 
+        internal abstract object TryGetService(Type serviceType);
```

Comment:
With A Try, I expected a bool resturn and a out param.

Proposal: rename to FindService

Reply:
it's a common naming convention for me (find returns `null` when it's not there, get thows an exception for that case).

 See also https://softwareengineering.stackexchange.com/a/182120/16953

Reply:
Yes thank you for spotting. Fixed TryGetService so it returns true / false